builtin/k8s/platform.go

```diff
@@ -698,20 +702,25 @@ func (p *Platform) resourceDeploymentCreate(
 		reportedError bool
 	)
 
-	timeout := 10 * time.Minute
+	// We wait the maximum amount of time that the deployment controller would wait for a pod
+	// to start before exitting. We double the time to allow for various kubernetes based
+	// delays in startup, detection, and reporting.
+	timeout := time.Duration((timeoutSeconds*failureThreshold)+initialDelaySeconds) * 2 * time.Second
+
+	podsSeen := make(map[types.UID]string)
 
 	// Wait on the Pod to start
-	err = wait.PollImmediate(2*time.Second, timeout, func() (bool, error) {
+	err = wait.PollImmediate(time.Second, timeout, func() (bool, error) {
 		dep, err := dc.Get(ctx, result.Name, metav1.GetOptions{})
 		if err != nil {
 			return false, err
 		}
 
 		if time.Since(lastStatus) > 10*time.Second {
 			step.Update(fmt.Sprintf(
-				"Waiting on deployment to become available: %d/%d/%d",
+				"Waiting on deployment to become available: requested=%d running=%d ready=%d",
```

Comment:
I noticed this change in your demo and I like it! 👍🏻 ✨

Reply:
![bitmoji](https://sdk.bitmoji.com/render/panel/881fa2bd-d9dd-4ab1-859e-5a70e95f4a70-8cb64c44-6a64-4f62-9642-460dd37039d6-v1.png?transparent=1&palette=1&width=246)